homeassistant/components/knx/config_flow.py

```diff
@@ -193,6 +219,85 @@ async def async_step_manual_tunnel(
             step_id="manual_tunnel", data_schema=vol.Schema(fields), errors=errors
         )
 
+    async def async_step_secure_manual(
+        self, user_input: dict | None = None
+    ) -> FlowResult:
+        """Configure ip secure manually."""
+        errors: dict = {}
+
+        if user_input is not None:
+            assert self._tunneling_config
+            entry_data: KNXConfigEntryData = {
+                **self._tunneling_config,  # type: ignore[misc]
+                CONF_KNX_SECURE_USER_ID: user_input[CONF_KNX_SECURE_USER_ID],
+                CONF_KNX_SECURE_USER_PASSWORD: user_input[
+                    CONF_KNX_SECURE_USER_PASSWORD
+                ],
+                CONF_KNX_SECURE_DEVICE_AUTHENTICATION: user_input[
+                    CONF_KNX_SECURE_DEVICE_AUTHENTICATION
+                ],
+                CONF_KNX_CONNECTION_TYPE: CONF_KNX_TUNNELING_TCP_SECURE,
+            }
+
+            return self.async_create_entry(
+                title=f"Secure {CONF_KNX_TUNNELING.capitalize()} @ {self._tunneling_config[CONF_HOST]}",
+                data=entry_data,
+            )
+
+        fields = {
+            vol.Required(CONF_KNX_SECURE_USER_ID): int,
+            vol.Required(CONF_KNX_SECURE_USER_PASSWORD): str,
+            vol.Required(CONF_KNX_SECURE_DEVICE_AUTHENTICATION): str,
+        }
+
+        return self.async_show_form(
+            step_id="secure_manual", data_schema=vol.Schema(fields), errors=errors
+        )
+
+    async def async_step_secure_knxkeys(
+        self, user_input: dict | None = None
+    ) -> FlowResult:
+        """Configure secure knxkeys used to authenticate."""
```

Comment:
knxkeys? Is that a word?

Reply:
its a file extension `my_project.knxkeys`. Shall we put it in quotes?

Reply:
Ok.